### PR TITLE
feat: add distro crate [fixes NET-463]

### PR DESCRIPTION
**Pull request by kmd-fl**

### CODE REVIEW

**Comment by kmd-fl** [comment]
@nahsi can you help add automatic distro crate publishing?

File: .github/release-please/config.json [hunk 1]
```diff
@@ -31,6 +32,9 @@
     },
     "keypair": {
       "component": "keypair"
+    }.
```
Comment (folex) [inline .github/release-please/config.json:35]: typo?

**Review by folex — COMMENTED** [review]
I don't understand a few things:
- What about builtin-package directory? You shouldn't need it anymore
- Why trust-graph config is in toml? shouldn't it be in json?

**Comment by kmd-fl** [comment]
@folex, replying to
> Why trust-graph config is in toml? shouldn't it be in json?

It's in toml because nox [accepts](https://github.com/fluencelabs/nox/blob/master/particle-modules/src/files.rs#L76) modules configs in toml

It's not clear why should it be in json at all.

**Comment by folex** [comment]
> @folex, it looks like it's difficult to fully eliminate `builtin-package` file atm since it's a part of CI, I'm not sure what's the exact process should be and how to transit to the new system without many problems

Makes sense. Then it's ok to have a semi-transitioned repo for a few releases. So let's keep builtin-package and let's remove it once we have finished with nox releases with new builtins system

**Comment by kmd-fl** [comment]
In order to fix CI I
1. Excluded distro package from `cargo nextest`. For this, I modified workflow files. 
2. Used `cfg` macro in distro/src/lib.rs to generate empty constants (not reading from FS) for clippy to be able to check the files
3. Now, in the repo, there are two Config.toml files that configure registry: in `service/` and in `distro/` dirs. `distro/Config.toml` is compatible with the rust-peer, but can't be used for tests because of some marine errors. On the other hand. `service/Config.toml` can be used for tests but isn't compatible with the node. I don't know how to fix this properly, so for now, I just created two versions. Need to fix this soon.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -42,7 +42,7 @@ jobs:
         env:
           NEXTEST_RETRIES: 10
           NEXTEST_TEST_THREADS: 10
-        run: cargo nextest run --release --all-features --no-fail-fast
+        run: cargo nextest run --release --all-features --no-fail-fast --workspace --exclude trust-graph-distro
```
Comment (folex) [inline .github/workflows/tests.yml:45]: why exclude?

Comment (folex) [reply .github/workflows/tests.yml:45]: can you please explain it in a comment?

**Review by justprosh — APPROVED** [review]

**Pull request merged**